lib/web/join_tokens_test.go

```diff
@@ -46,6 +46,231 @@ func TestCreateNodeJoinToken(t *testing.T) {
 	require.Equal(t, "some-token-id", token.ID)
 }
 
+func TestGenerateIamTokenName(t *testing.T) {
+	rule1 := types.TokenRule{
+		AWSAccount: "100000000000",
+		AWSARN:     "arn:aws:iam:1",
+	}
+
+	rule2 := types.TokenRule{
+		AWSAccount: "200000000000",
+		AWSARN:     "arn:aws:iam:b",
+	}
+
+	// make sure the order doesn't matter
+	hash1, err := generateIamTokenName([]*types.TokenRule{&rule1, &rule2})
+	require.NoError(t, err)
+
+	hash2, err := generateIamTokenName([]*types.TokenRule{&rule2, &rule1})
+	require.NoError(t, err)
+
+	require.Equal(t, hash1, hash2)
+
+	// generate different hashes for different rules
+	hash1, err = generateIamTokenName([]*types.TokenRule{&rule1})
+	require.NoError(t, err)
+
+	hash2, err = generateIamTokenName([]*types.TokenRule{&rule2})
+	require.NoError(t, err)
+
+	require.NotEqual(t, hash1, hash2)
+
```

Comment:
Can you add a test that the hashes should not change over time in case someone accidentally changes the hash algorithm or adds a hyphen or something like I mentioned above? Could compare the hash of one of these rule sets to its current known value.

Reply:
Thanks, added in https://github.com/gravitational/teleport/pull/11339/commits/2fea8f1a6967cf534610de725c1b767bcfd7e737